source/material_model/visco_plastic.cc

```diff
@@ -187,7 +187,10 @@ namespace aspect
           // Step 1b: compute viscosity from dislocation creep law
           const double viscosity_dislocation = dislocation_creep.compute_viscosity(edot_ii, pressure, temperature_for_viscosity, j);
 
-          // Step 1c: select what form of viscosity to use (diffusion, dislocation or composite)
+          // Step 1c: compute viscosity from Frank-Kamenetskii approximation
+          const double viscosity_frank_kamenetskii = frank_kamenetskii.compute_viscosity(ref_visc, temperature_for_viscosity, j);
```

Comment:
Please do not use `ref_visc` here. It is a variable that is currently only used for internal purposes and will go away (it is for example not used for the diffusion or dislocation creep). Instead let the Frank-Kamenetskii plugin read in the prefactors.

Reply:
Also as we discussed today we actually do not want to allow an adiabatic temperature gradient for the Frank Kamenetskii viscosity, so the temperature handed over should be `temperature`, not `temperature_for_viscosity`. Do I remember that correctly?

Reply:
@gassmoeller @erinheilman - yes, `temperature` should be handed over to the Frank-Kamenetskii rheology module, rather than `temperature_for_viscosity`.